ui/lib/provider-credentials/build-crendentials.ts

```diff
@@ -197,10 +197,49 @@ export const buildGitHubSecret = (formData: FormData) => {
   return {};
 };
 
+export const buildOracleCloudSecret = (formData: FormData, providerUid?: string) => {
+  const keyContent = getFormValue(
+    formData,
+    ProviderCredentialFields.OCI_KEY_CONTENT,
+  ) as string;
+
+  // Base64 encode the key content for the backend
+  // Use btoa with proper UTF-8 encoding to handle all characters
+  const encodedKeyContent = keyContent
+    ? btoa(unescape(encodeURIComponent(keyContent)))
+    : "";
```

Comment:
The combination of btoa(unescape(encodeURIComponent())) is a legacy workaround that uses deprecated APIs (unescape is deprecated).

```suggestion
const encodedKeyContent = keyContent
    ? btoa(String.fromCharCode(...new TextEncoder().encode(keyContent)))
    : "";

  // OR use a utility function for base64 encoding
  const base64Encode = (str: string): string => {
    return btoa(String.fromCharCode(...new TextEncoder().encode(str)));
  };
```